fix(intro): make :help news line easier to translate

Prince213 opened the pull request:
This PR introduces two changes:
1. `"https://neovim.io/#chat"` is no longer marked as translatable. ([ref](https://www.gnu.org/software/gettext/manual/html_node/Preparing-Strings.html#No-embedded-URLs))
2. The `:help news<Enter>` line can now be properly translated for `zh_CN.UTF-8`. Previous approach makes it difficult to translate because of the word order.

Prince213 committed:
fix(i18n): don't translate URL

Prince213 committed:
fix(i18n): make the intro message translatable

Prince213 committed:
feat(l10n): update zh_CN.UTF-8 translations

zeertzjq commented on src/nvim/version.c:
```diff
@@ -2834,24 +2836,39 @@ void intro_message(int colon)
-        } else if (strstr(p, "iccf") != NULL) {
-          p = sponsor < 0
+          p = NULL;
+        }
+        if (strstr(p, "iccf") != NULL) {
+          mesg = sponsor < 0
               ? N_("type  :help sponsor<Enter>    for information ")
               : N_("type  :help register<Enter>   for information ");
-        } else if (strstr(p, "Orphans") != NULL) {
-          p = N_("menu  Help->Sponsor/Register  for information    ");
+          p = NULL;
         }
       }
 
-      if (*p != NUL) {
-        do_intro_line(row, _(p), 0);
+      if (*mesg != NUL) {
+        do_intro_line(row, mesg, 0);
```
This doesn't look right. With this change the `:help news` line is translated, but the other lines are not.

Prince213 committed:
fix(i18n): fix lines don't get translated

Prince213 committed:
fix: fix cast

Prince213 committed:
fix: handle empty line properly

Previous implementation passes empty line directly to gettext, which
causes it to return the header entry, causing the output to be messed
up. Add a special check for this case.

zeertzjq commented on src/nvim/version.c:
```diff
@@ -2833,25 +2835,48 @@ void intro_message(int colon)
+          mesg = sponsor < 0
+              ? _("Sponsor Vim development!")
+              : _("Become a registered Vim user!");
+        }
+        if (strstr(p, "iccf") != NULL) {
```
```suggestion
        } else if (strstr(p, "iccf") != NULL) {
```

zeertzjq commented on runtime/doc/news.txt:
```diff
@@ -167,6 +167,8 @@ The following changes to existing APIs or features add new behavior.
 
 • API calls now show more information about where an exception happened.
 
+• Strings in :intro are now friendlier to translators.
```
news.txt is introduced during 0.9 release cycle, so this PR doesn't need a news.txt update. It's fine to just let the new.txt workflow fail.

zeertzjq commented on src/nvim/version.c:
```diff
@@ -2833,25 +2835,48 @@ void intro_message(int colon)
-              : N_("type  :help register<Enter>   for information ");
-        } else if (strstr(p, "Orphans") != NULL) {
-          p = N_("menu  Help->Sponsor/Register  for information    ");
+          mesg = sponsor < 0
+              ? _("Sponsor Vim development!")
+              : _("Become a registered Vim user!");
+        }
+        if (strstr(p, "iccf") != NULL) {
+          mesg = sponsor < 0
+              ? _("type  :help sponsor<Enter>    for information ")
+              : _("type  :help register<Enter>   for information ");
+        }
```
The change from `p` to `mesg` isn't necessary here now as it is translated below. Removing the unused message is enough.
```suggestion
          p = sponsor < 0
              ? N_("Sponsor Vim development!")
              : N_("Become a registered Vim user!");
        } else if (strstr(p, "iccf") != NULL) {
          p = sponsor < 0
              ? N_("type  :help sponsor<Enter>    for information ")
              : N_("type  :help register<Enter>   for information ");
        }
```
